tests/ops/functions/test_commutator.py

```diff
@@ -80,22 +81,24 @@ def test_no_recording_in_context_with_pauli():
     with qml.tape.QuantumTape() as tape2:
         qml.PauliX(0)
 
+    assert len(tape) == len(tape2) == 1
     qml.assert_equal(tape, tape2)
 
 
+@pytest.mark.xfail(strict=True)
```

Comment:
This test relies on `s_prod` being queued (and was passing wrongly because it was actually not queued in either of the two tapes...)

Reply:
> Can we then update this test to instead test the behavior we actually want?

I thought this way is nicer because now it is a test for how we want things, but there is a deviation in behaviour before merging #8131 that is marked by `xfail`.
Basically the desired behaviour is not available in PennyLane until merging that PR, unless fiddling manually with the queue.

Reply:
> Also, I know this might be irrelavant to this PR, but can we update these tests to use AnnotatedQueue instead of QuantumTape? 

Will do!

Reply:
Actually, when using `AnnotatedQueue` instead of `QuantumTape`, the problem disappears; I'm confused.

Reply:
Ah, right. "queuing" (put into a queue) of `SProd` (or objects with `_queue_category=None` in general) happens always. It is only when we process the queue into a tape that such an object is ignored/not "queued" (put into a tape).